compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -215,7 +215,8 @@ pub fn check_intrinsic_type(tcx: TyCtxt<'_>, it: &hir::ForeignItem<'_>) {
 
             sym::type_name => (1, Vec::new(), tcx.mk_static_str()),
             sym::type_id => (1, Vec::new(), tcx.types.u64),
-            sym::offset | sym::arith_offset => (
+            sym::offset => (2, vec![param(0), param(1)], param(0)),
+            sym::arith_offset => (
```

Comment:
Is `arith_offset` not changed because we don't have a version of it in MIR?

Makes me wander if we should try intrinsicing other offset like functions like `{byte,wrapping,wrapping_byte}_{add,sub,offset}`

Reply:
@WaffleLapkin The weird thing about `byte_add` is that it works on fat pointers.  So it ends up quite a bit more complex than `offset`.  I'm not sure if that's worth doing as an intrinsic or not.  For the data part of it, cast-then-`offset`-then-cast of course works fine.

As for `sub`, maybe we should start by changing it from `wrapping_neg` to `unchecked_neg`, especially after #112238...

Reply:
Hm, but wouldn't `byte_add` be just `self.data += x`? Doesn't sound like it's too hard for an intrinsic...

Reply:
@scottmcm I cleared up old notifications that I were putting off for a long time 😄